runtime/src/iree-amd-aie/aie_runtime/xaie_hwcfg.c

```diff
@@ -0,0 +1,166 @@
+const int XAIE1_NUM_NOC_INTR_OFFSET = XAIE_NUM_NOC_INTR_OFFSET;
+
+// TODO(max): inconsistent use of XAIE_BASE_ADDR when passed to
+// XAie_SetupPartitionConfig and XAie_SetupConfig
+// https://github.com/Xilinx/aie-rt/blob/203e61708e683cc8cd62205376ea19bb9f5a6233/driver/examples/xaie_intr_cdo_generate.c#L12
+const uint64_t XAIE1_PARTITION_BASE_ADDR = XAIE_BASE_ADDR;
+
+#undef XAIE_BASE_ADDR
+#undef XAIE_NPI_BASEADDR
+#undef XAIE_NUM_ROWS
+#undef XAIE_NUM_COLS
+#undef XAIE_TILE_NUM_LOCKS
+#undef XAIE_SHIM_NUM_LOCKS
+#undef XAIE_TILE_DMA_NUM_CH
+#undef XAIE_SHIM_DMA_NUM_CH
+#undef XAIE_COL_SHIFT
+#undef XAIE_ROW_SHIFT
+#undef XAIE_SHIM_ROW
+#undef XAIE_SHIM_NUM_ROWS
+#undef XAIE_MEM_TILE_ROW_START
+#undef XAIE_MEM_TILE_NUM_ROWS
+#undef XAIE_AIE_TILE_ROW_START
+#undef XAIE_AIE_TILE_NUM_ROWS
+#undef XAIE_NUM_NOC_INTR_OFFSET
+#undef XAIE_CORE_TILE_NUM_EVENT_STATUS_REGS
+#undef XAIE_SHIM_TILE_NUM_EVENT_STATUS_REGS
+#undef XAIE_MEM_TILE_NUM_EVENT_STATUS_REGS
+#undef XAIE_DEV_SINGLE_GEN
+#undef XAIE_LITE_HWCFG_H
+
+// AIE Gen ML
+#define XAIE_DEV_SINGLE_GEN XAIE_DEV_GEN_AIEML
+#include "xaiengine/xaie_lite_hwcfg.h"
+
+const uint64_t XAIEML_BASE_ADDR = XAIE_BASE_ADDR;
+const uint64_t XAIEML_NPI_BASEADDR = XAIE_NPI_BASEADDR;
+const int XAIEML_NUM_ROWS = XAIE_NUM_ROWS;
+const int XAIEML_NUM_COLS = XAIE_NUM_COLS;
+const int XAIEML_TILE_NUM_LOCKS = XAIE_TILE_NUM_LOCKS;
+const int XAIEML_MEM_TILE_NUM_LOCKS = XAIE_MEM_TILE_NUM_LOCKS;
+const int XAIEML_SHIM_NUM_LOCKS = XAIE_SHIM_NUM_LOCKS;
+const int XAIEML_TILE_DMA_NUM_CH = XAIE_TILE_DMA_NUM_CH;
+const int XAIEML_MEM_TILE_DMA_NUM_CH = XAIE_MEM_TILE_DMA_NUM_CH;
+const int XAIEML_SHIM_DMA_NUM_CH = XAIE_SHIM_DMA_NUM_CH;
+const int XAIEML_COL_SHIFT = XAIE_COL_SHIFT;
+const int XAIEML_ROW_SHIFT = XAIE_ROW_SHIFT;
+const int XAIEML_SHIM_ROW = XAIE_SHIM_ROW;
+const int XAIEML_SHIM_NUM_ROWS = XAIE_SHIM_NUM_ROWS;
+const int XAIEML_MEM_TILE_ROW_START = XAIE_MEM_TILE_ROW_START;
+const int XAIEML_MEM_TILE_NUM_ROWS = XAIE_MEM_TILE_NUM_ROWS;
+const int XAIEML_AIE_TILE_ROW_START = XAIE_AIE_TILE_ROW_START;
+const int XAIEML_AIE_TILE_NUM_ROWS = XAIE_AIE_TILE_NUM_ROWS;
+const int XAIEML_NUM_NOC_INTR_OFFSET = XAIE_NUM_NOC_INTR_OFFSET;
+
+const uint64_t XAIEML_PARTITION_BASE_ADDR = XAIE_BASE_ADDR;
+
+#undef XAIE_BASE_ADDR
+#undef XAIE_NPI_BASEADDR
+#undef XAIE_NUM_ROWS
+#undef XAIE_NUM_COLS
+#undef XAIE_TILE_NUM_LOCKS
+#undef XAIE_SHIM_NUM_LOCKS
+#undef XAIE_TILE_DMA_NUM_CH
+#undef XAIE_SHIM_DMA_NUM_CH
+#undef XAIE_COL_SHIFT
+#undef XAIE_ROW_SHIFT
+#undef XAIE_SHIM_ROW
+#undef XAIE_SHIM_NUM_ROWS
+#undef XAIE_MEM_TILE_ROW_START
+#undef XAIE_MEM_TILE_NUM_ROWS
+#undef XAIE_AIE_TILE_ROW_START
+#undef XAIE_AIE_TILE_NUM_ROWS
+#undef XAIE_NUM_NOC_INTR_OFFSET
+#undef XAIE_CORE_TILE_NUM_EVENT_STATUS_REGS
+#undef XAIE_SHIM_TILE_NUM_EVENT_STATUS_REGS
+#undef XAIE_MEM_TILE_NUM_EVENT_STATUS_REGS
+#undef XAIE_DEV_SINGLE_GEN
+#undef XAIE_LITE_HWCFG_H
+
+// AIE Gen IPU
+#define IPU_AIE_BASEADDR 0x40000000;
+#define IPU_AIE_NPI_ADDR 0x0;
```

Comment:
Are these needed? Can't the existing `XAIE2IPU_BASE_ADDR` and `XAIE2IPU_NPI_BASEADDR` be reused?

Reply:
Well `XAIE2IPU*` don't exist (those are just my names for these things). But `XAIE_BASE_ADDR`, `XAIE_NPI_BASEADDR` for IPU don't exist either because I can't find where some key headers are stashed https://github.com/Xilinx/aie-rt/blob/release/main_aig/driver/src/lite/xaie_lite_hwcfg.h#L113-L123

```
#ifdef __AIEIPU__
#include <platform-hw-config.h>
#endif

#ifndef XAIE_BASE_ADDR
#define XAIE_BASE_ADDR			IPU_AIE_BASEADDR
#endif

#ifndef XAIE_NPI_BASEADDR
#define XAIE_NPI_BASEADDR		IPU_AIE_NPI_ADDR
#endif
```

`platform-hw-config.h` is in [XRT](https://github.com/Xilinx/XRT/blob/master/src/runtime_src/tools/xclbinutil/aie-pdi-transform/libinclude/platform-hw-config.h#L10) (:facepalm:)

```
#ifdef _ENABLE_IPU_LX6_
#include <ipu.h>
extern aie2_manager_t aie2_manager_info;
```

but I can't find `ipu.h` (and git blame person hasn't responded).

Reply:
I see, strange where that file comes from.. but thanks for the explanation.